pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -103,6 +103,19 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 					break
 				}
 			}
+			// If RevisionConditionResourcesAvailable is false, with reason being "PodInitializing",
+			// check to see if the secrets are fetchable
+			if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionResourcesAvailable); cond != nil && cond.Status == corev1.ConditionFalse && cond.Reason == "PodInitializing" {
+				for _, volume := range rev.Spec.Volumes {
+					if secret := volume.VolumeSource.Secret; secret != nil {
+						_, err := c.secretLister.Secrets(rev.Namespace).Get(secret.SecretName)
```

Comment:
Why check this only after we get the failure? Do we think the performance impact is high for Revision creation? If we check this before we create the Pod we could return the error much faster to the user and do a whole lot less work on the cluster.

Reply:
Right, instead of waiting for it to timeout, we should do this check before pods are created